src/main/scala/firrtl/passes/memlib/MemIR.scala

```diff
@@ -0,0 +1,43 @@
+// See LICENSE for license details.
+
+package firrtl.passes
+package memlib
+
+import firrtl._
+import firrtl.ir._
+import Utils.indent
+
+case class DefAnnotatedMemory(
+    info: Info,
+    name: String,
+    dataType: Type,
+    depth: Int,
+    writeLatency: Int,
+    readLatency: Int,
+    readers: Seq[String],
+    writers: Seq[String],
+    readwriters: Seq[String],
+    readUnderWrite: Option[String],
+    maskGran: Option[BigInt],
+    memRef: Option[String]
+    //pins: Seq[Pin],
+    ) extends Statement with IsDeclaration {
+  def serialize: String =
```

Comment:
So serialization on intermediate passes is nice for debugging. Once we have the annotation system a bit more complete we can serialize them as well, but for now I think it's fine for the thing to just serialize as a defmem. In fact rather than reimplementing the code from defmem's serialize, I'd just create one and call serialize on that. ie.

```scala
def serialize: String = DefMemory(info, name, dataType, depth,...).serialize
```

I think that would show the intent here better, and since serialize is constructing strings anyway, I'm sure creating an extra short-lived object is irrelevant for performance.